onnxruntime/server/core/metric_registry.h

```diff
@@ -0,0 +1,113 @@
+#pragma once
+
+#include <prometheus/registry.h>
+
+namespace onnxruntime {
+namespace server {
+struct MetricRegistry {
+  // returns the global metric registry instance
+  // use this handle to get the instance to increment/count requests
+  static MetricRegistry& Get() {
+      static MetricRegistry metricRegistry;
+      return metricRegistry;
+  }
+
+  // Store quantile buckets 
+  static prometheus::Histogram::BucketBoundaries TimeBuckets () {
+    // Anything greater than 10s we should bucket into +inf
+    return {0, 50, 100, 250, 500, 1000, 1000, 2500, 5000, 10000};
+  }
+
+  static prometheus::Histogram::BucketBoundaries ByteBuckets() {
+    // get megabyte boundaries?
+    return {
+      0, 500000, 1000000, 2000000, 3000000, 4000000,
+      5000000, 6000000, 7000000, 8000000, 9000000, 10000000
+    };
+  }
+
+  // The internal prometheus registry
+  prometheus::Registry registry;
+  // The three metrics we're currently recording
+  // Inference Time (model and version can be handled by labels so you can)
+  // aggregate total onnxruntime server performance if desired
+  std::shared_ptr<prometheus::Family<prometheus::Histogram>*> inferenceTimer;
```

Comment:
Why not use `std::shared_ptr<prometheus::Family<prometheus::Histogram>> inferenceTimer;`?

Reply:
The mutex inside Family prevent the copy constructor for the shared pointer. This was the workaround.

Reply:
Consider the object here or `unique_ptr`? I could not find anywhere using it other than `Get()`.